### PR TITLE
Fixed CopyOutputCall for scattered calls

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

**Comment by cjllanwarne** [comment]
Wheelers: @Horneth @scottfrazer

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 1]
```diff
@@ -716,9 +716,18 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       (upstream forall { _.nonEmpty }) && dependenciesResolved
     }
 
+    lazy val nonFinalCallsComplete= executionStore forall {
```
Comment (Horneth) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: space before =

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 2]
```diff
@@ -716,9 +716,18 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
...
+      case (key: FinalCallKey, _) => true
+      case (_, x: ExecutionStatus) if x.isTerminal => true
+      case _ => false
+    }
+
     def isRunnable(entry: ExecutionStoreEntry) = {
-      val (key, status) = entry
-      status == ExecutionStatus.NotStarted && arePrerequisitesDone(key)
+      entry match {
+        case (key: FinalCallKey, ExecutionStatus.NotStarted) => nonFinalCallsComplete
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: Thanks for fixing this.  What I tried (and failed) to do was make a final call dependent on every other call.  I think I must have confused the scatter placeholder with the collector; the latter is only generated after the scatter explodes, and the former is not sufficient for the final call to depend on.

Anyhoo there is a misleading comment above `prerequisiteScopes` in `FinalCall` that should be corrected.  I'm less certain what do about the code.

Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: I updated the comment and removed the (no-longer-used) dependencies

**Comment by geoffjentry** [comment]
:+1:

**Comment by Horneth** [comment]
:+1:

**Pull request merged**